File: Edward Poot/src/QL/ast/expr/binaryoperatorexpr/geq.go
package binaryoperatorexpr

import ("ql/ast/visit"
"ql/ast/expr")

type GEq struct {
	Lhs, Rhs expr.Expr
}

func (g GEq) GetLhs() expr.Expr {
	return g.Lhs
}

func (g GEq) GetRhs() expr.Expr {
	return g.Rhs
}

func (g GEq) Eval() interface{} {
	return g.GetLhs().Eval().(int) >= g.GetRhs().Eval().(int)
}

func (g GEq) Accept(v visit.Visitor) interface{} {
	return v.Visit(g)
}

File: Kevin van den Bekerom/DSLQL/src/main/java/org/uva/sea/ql/ast/expr/GEq.java
package org.uva.sea.ql.ast.expr;

import org.uva.sea.ql.ast.Visitable;
import org.uva.sea.ql.ast.Visitor;

public class GEq extends BinaryExpr implements Visitable {
	
	public GEq(Expr lhs, Expr rhs) {
		super.lhs = lhs;
		super.rhs = rhs;
		super.type = Type.BOOLEAN;
	}
	
	@Override
	public Boolean eval() {
		return (Integer) lhs.eval() >= (Integer) rhs.eval();
	}
	
	public void accept(Visitor visitor) {
		visitor.visit(this);
	}
}